Issue #3224859 by nkoporec: Flexible groups with visibility set to secret, should not show up in views results

nkoporec opened the pull request:
## Problem
When I'm logged in as an authenticated user and I go to /all-groups page, I see groups that has the visibility option set to 'Group members only (secret)' even though I'm not part of that group.

## Solution
We need to alter the views query, so that it will only show secret groups that the user is part of.

## Issue tracker
https://www.drupal.org/project/social/issues/3224859

## How to test
1. As a site manager create a group with the visibility set to 'Group members only (secret)'
2. Log in as authenticated user and visit /all-groups page.
3. See that the secret group is displayed, even though I'm not a member of it.

## Screenshots
Before:
![2021-07-22_12-10](https://user-images.githubusercontent.com/35064680/126624465-4f26f856-9436-4ee6-bc9f-ad24489821d2.png)

After:
![2021-07-22_12-08](https://user-images.githubusercontent.com/35064680/126624491-4d98e45d-f1c1-4d60-91b0-0376d9d02c5e.png)


## Release notes
Fixed an issue where groups with visibility set to 'Group members only (secret)' would still show up in the views results.

## Change Record


## Translations

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2432-cqqury4mjype023xhw8hcglr20src44r.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60f9464d7424684318654948

navneet0693 commented:
@nkoporec The few other test failures might be false but the "group create flexible" test is failing, https://app.travis-ci.com/github/goalgorilla/open_social/jobs/526173256#L2280. Do you think this is correct?

nkoporec commented:
> @nkoporec The few other test failures might be false but the "group create flexible" test is failing, https://app.travis-ci.com/github/goalgorilla/open_social/jobs/526173256#L2280. Do you think this is correct?

Yes, the test was indeed failing. The issue was with the anonymous user, I have removed all the logic that altered the query  for anonymous users, as it is no longer  needed due to my changes of the member query. I have also removed the secret_group  query, since it was already done in the social_secret_group module and there is no need to repeat the query.

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2432-cqqury4mjype023xhw8hcglr20src44r.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60f9464d7424684318654948

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2432-cqqury4mjype023xhw8hcglr20src44r.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60f9464d7424684318654948

robertragas commented on modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.module:
```diff
@@ -694,70 +694,37 @@ function social_group_flexible_group_views_query_alter(ViewExecutable $view, Que
-    $query->addWhere($new_where, 'field_flexible_group_visibility_value', ['public'], 'IN');
-    $query->addWhere($new_where, 'groups_field_data.type', ['flexible_group'], 'IN');
-    return;
+    $query->addWhere($new_where, 'id', $ids, 'NOT IN');
```
Think we need to change this to 
$query->addWhere($new_where, 'groups_field_data.id', $ids, 'NOT IN');

Because this query is making a lot joins, and ends up in this id being ambiguous.

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2432-cqqury4mjype023xhw8hcglr20src44r.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60f9464d7424684318654948

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2432-cqqury4mjype023xhw8hcglr20src44r.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60f9464d7424684318654948

robertragas commented:
Merging the PR since we processed the last change request. The failing travis tests are unrelated.
